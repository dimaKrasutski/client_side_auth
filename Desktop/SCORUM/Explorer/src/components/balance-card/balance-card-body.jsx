import React, { Fragment } from 'react';
import PropTypes from 'prop-types';
import { FormattedMessage } from 'react-intl';
import { formatBalanceWithLocale, getBalanceConverter } from '../../helpers/accounts';
import { currencyPropTypes } from '../../helpers/prop-types/currency';

export const BalanceCardBody = ({
  icon, title, balance, locale, currency,
}) => {
  const rate = 1;
  const formatBalanceLocale = formatBalanceWithLocale(locale);
  const convertToCurrency = getBalanceConverter(rate);
  return (
    <Fragment>
      <span className="d-flex align-items-center pb-1_5">
        {icon}
        <span className="ml-1 font-weight-normal">
          <FormattedMessage id={title} />
        </span>
      </span>
      <div>
        <h2 className="mb-0 h2">
          {balance ?
            formatBalanceLocale(balance) :
            <div className="preloader mt-1_5 mb-2 preloader-lg" />
          }
        </h2>
        <h4 className="text-gray-dark-hint">
          {balance && rate ?
            `${formatBalanceLocale(convertToCurrency(balance))} ${currency}` :
            <div className="preloader mb-1_5" />
          }
        </h4>
      </div>
    </Fragment>
  );
};

BalanceCardBody.propTypes = {
  icon: PropTypes.node.isRequired,
  title: PropTypes.string.isRequired,
  balance: PropTypes.number,
  currency: currencyPropTypes.isRequired,
  locale: PropTypes.string.isRequired,
};

BalanceCardBody.defaultProps = {
  balance: null,
};
