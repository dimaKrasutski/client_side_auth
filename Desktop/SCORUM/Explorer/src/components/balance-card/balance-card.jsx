import React from 'react';
import classnames from 'classnames';
import PropTypes from 'prop-types';
import Input from 'reactstrap/lib/Input';

export const BalanceCard = ({ classNames, innerClassName, children }) => (
  <div className={classNames}>
    <Input tag="div" type="textarea" className={classnames(innerClassName, 'plain-textarea')} plaintext bsSize="lg">
      {children}
    </Input>
  </div>
);

BalanceCard.propTypes = {
  classNames: PropTypes.string,
  innerClassName: PropTypes.string,
  children: PropTypes.node.isRequired,
};

BalanceCard.defaultProps = {
  classNames: null,
  innerClassName: null,
};
