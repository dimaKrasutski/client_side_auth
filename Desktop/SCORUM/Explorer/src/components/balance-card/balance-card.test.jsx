import React from 'react';
import { BalanceCard } from './balance-card';

describe('<BalanceCard />', () => {
  test('should render balance card', () => {
    const wrapper = shallow(<BalanceCard classNames="test">Some info about balance</BalanceCard>);
    expect(wrapper).toMatchSnapshot();
  });
});
