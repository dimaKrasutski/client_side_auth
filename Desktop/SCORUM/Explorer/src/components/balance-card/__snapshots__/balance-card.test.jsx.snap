// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<BalanceCard /> should render balance card 1`] = `
<div
  className="test"
>
  <Input
    bsSize="lg"
    className="plain-textarea"
    plaintext={true}
    tag="div"
    type="textarea"
  >
    Some info about balance
  </Input>
</div>
`;
